docs/guides/server/hostname.adoc

```diff
@@ -3,141 +3,127 @@
 
-<@kc.start parameters="--hostname-admin=<host>"/>
+WARNING: If either `forwarded` or `xforwarded` is selected, make sure your reverse proxy properly sets and overwrites the `Forwarded` or `X-Forwarded-*` headers respectively. To set these headers, consult the documentation for your reverse proxy. Misconfiguration will leave {project_name} exposed to security vulnerabilities.
 
-However, if you want to set not only the host but also a scheme, port, and path, you can set the `hostname-admin-url` option:
+== Exposing the Administration Console on a separate hostname
 
-<@kc.start parameters="--hostname-admin-url=<scheme>://<host>:<port>/<path>"/>
+If you wish to expose the Admin Console on a different host, you can do so with the following command:
 
-Note that the `hostname-admin` and `hostname-admin-url` are mutually exclusive.
+<@kc.start parameters="--hostname https://my.keycloak.org --hostname-admin https://admin.my.keycloak.org:8443"/>
 
-To reduce attack surface, the administration endpoints for {project_name} and the Admin Console should not be publicly accessible.
-Therefore, you can secure them by using a reverse proxy.
-For more information about which paths to expose using a reverse proxy, see <@links.server id="reverseproxy"/>.
+This allows you to access {project_name} at `https://my.keycloak.org` and the Admin Console at `https://admin.my.keycloak.org:8443`, while the backend continues to use `https://my.keycloak.org`.
 
-== Example Scenarios
-The following are more example scenarios and the corresponding commands for setting up a hostname.
 
-Note that the `start` command requires setting up TLS. The corresponding options are not shown for example purposes. For more details, see <@links.server id="enabletls"/>.
+NOTE: Keep in mind that hostname and proxy options do not change the ports on which the server listens. Instead it changes only the ports of static resources like JavaScript and CSS links, OIDC well-known endpoints, redirect URIs, etc. that will be used in front of the proxy. You need to use HTTP configuration options to change the actual ports the server is listening on. Refer to the <@links.server id="all-config"/> for details.
 
-=== Exposing the server behind a TLS termination proxy
+== Background - server endpoints
 
-In this example, the server is running behind a TLS termination proxy and publicly available from `https://mykeycloak`.
+{project_name} exposes several endpoints, each with a different purpose. They are typically used for communication among applications or for managing the server. We recognize 3 main endpoint groups:
 
-.Configuration:
-<@kc.start parameters="--hostname=mykeycloak --http-enabled=true --proxy-headers=forwarded|xforwarded"/>
+* Frontend
+* Backend
+* Administration
 
-=== Exposing the server without a proxy
+If you want to work with either of these endpoints, you need to set the base URL. The base URL consists of a several parts:
```

Comment:
@Pepo48 As we're stating here that the base URL should be set, IMHO, we should be more explicit and tell users HOW to do it. I like it in the old guide, as the behavior for all these endpoint types was briefly explained.

I'd comprehend a little bit the sub-types below with the examples. Specifying the `hostname`, `hostname-admin`, or just with the help of the dynamic resolutions,... I know for the admin, it's described much above, but I'd say for the other options, it's not so clear. 

In the sections below, there's mentioned only WHAT and WHY, but I'd also appreciate HOW - clearly. Like, how to easily set the backend endpoint? 

I'd appreciate having at least a brief comment for these sections with potential reference to the 'Sources for resolving URLs'. Or just combine them in some manner?

@Pepo48 WDYT?

Reply:
@mabartos this was caused by the new guide structure, where the emphasis is on the use-cases that goes first and then there is a little bit of background. The structure is the exact opposite of the old guide, where the background is mentioned at first. 
I felt that "HOW" is demonstrated enough in the first part, that's why the background part did not repeat it.

Anyway, I see your point - I added couple of sentences to each of the endpoints, while referring to the Validation section as well.

PTAL. Thanks.